edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/ModuleClient.cs

```diff
@@ -21,12 +23,32 @@ public class ModuleClient : IModuleClient
         static readonly RetryStrategy TransientRetryStrategy =
             new ExponentialBackoff(int.MaxValue, TimeSpan.FromSeconds(2), TimeSpan.FromSeconds(60), TimeSpan.FromSeconds(4));
 
-        static readonly IDictionary<UpstreamProtocol, TransportType> UpstreamProtocolTransportTypeMap = new Dictionary<UpstreamProtocol, TransportType>
+        static readonly IDictionary<UpstreamProtocol, Func<Option<IWebProxy>, ITransportSettings>> TransportMap = new Dictionary<UpstreamProtocol, Func<Option<IWebProxy>, ITransportSettings>>
```

Comment:
Nit: Having this as a static dictionary is a bit odd with the new changes. Suggest converting this to a static method instead that takes in UpstreamProtocol and Option<IWebProxy> and returns an ITransportSettings.

Reply:
Agreed, will do.

Reply:
Done.